src/Controller/OverviewController.php

```diff
@@ -222,6 +226,67 @@ public function update_category(Request $request, $id): Response
 
   }
 
+  /**
+   * This should be the main page that everyone should see. Every user should be able to see this page and everything
+   * on it. This will be modified more clearly from it's current state. Currently
+   * being used as a testing stage for database outputs.
+   * 
+   * @author Daniel Boling
+   * @return rendered moderation_announcements.html.twig
+   * 
+   * @Route("/moderation/announcements", name="moderation_announcements")
+   * @IsGranted("ROLE_MODERATOR")
+   */
+  public function moderation_announcements(): Response
+  {
+
+    $announcement = $this->getDoctrine()
+      // inits the database and table Announcements;
+      ->getRepository(Announcement::class)
+      ->find_today();
+
+      return $this->render('moderation_announcements.html.twig', [
+        'date' => $this->date,
+        'announcement' => $announcement,
+      ]);
+
+  }
+
+  /**
+   * Is called on button-click from twig file, updates active categories, and redirects to list_category
+   * 
+   * @author Daniel Boling
+   * @return redirect to list_category
+   * 
+   * @Route("/moderation/announcement/{id}", name="approve_announcement")
+   * @IsGranted("ROLE_MODERATOR")
+   */
+  public function approve_announcement(Request $request, $id): Response
+  {
+
+    $em = $this->getDoctrine()->getManager();
+
+    $announcement = $this->getDoctrine()
+      ->getRepository(Announcement::class)
+      ->find($id);
+      
+    if ($announcement->getApproval() == 0)
+    // if the announcement is denied, set it to approved.
+      {
+        $announcement->setApproval(1);
+
+      } else {
+      // if the condition gets here, the announcement is already approved, so set it to denied.
+        $announcement->setApproval(0);
+      }
```

Comment:
Your function is named `approve_announcement()` but this block un-approves the announcement.  Could you either remove this block so that if you call the function on an already approved announcement it doesn't remove the approval OR update the function name and description to reflect the fact that the function toggles the approval status rather than just approving it?  Does that make sense?

Reply:
Function name and description modified to better fit the purpose of the function, which was intended to serve as both approval and denial of announcements, assuming it was approved accidentally or needs removal.